grafana/grafana
# Explore: Fix time interpolation

**What this PR does / why we need it**:

When Explore is opened directly with `/explore` URL the interpolation of `__from` / `__to` variables doesn't work. This is because in such case `TemplateSrv` never receives `timeRange` that is used for interpolation. 

When a Dashboard page is opened first and then the user navigates to Explore the interpolation works fine, though it may use incorrect time range. This is because the time range applied by [this action](https://github.com/grafana/grafana/blob/main/public/app/features/variables/state/actions.ts#L134) becomes stale, never updated but still used for interpolation.

The fix is to update the current time range each time the range changes in `updateTime` action. Note that the flow is slightly different from Dashboards where the time range gets _refreshed_ in the `TemplateSrv` instance when [time picker changes](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx#L41). 

In Explore we call `updateTime` when a query is run or the date range changes. This also involves re-initializing `TimeSrv` (it had already been working this way), because we use **one singleton** used by **both Explore panes**. This looks like a code smell we need to look at later, though.

**Which issue(s) this PR fixes**:



Fixes https://github.com/grafana/support-escalations/issues/1459

**Special notes for your reviewer**:

How to test it?
1) Go to Explore
2) Use dev-loki
3) Run following query: `{ from="${__from}", to="${__to}" }`
4) Refresh Explore page
5) Check in the network tab that the acutal query that was passed to the backend has the interpolated time.
